cmd/sovereignnode/chainSimulator/tests/bridge/bridge_test.go

```diff
@@ -0,0 +1,221 @@
+package bridge
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	chainSimulatorIntegrationTests "github.com/multiversx/mx-chain-go/integrationTests/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	sovereignChainSimulator "github.com/multiversx/mx-chain-go/sovereignnode/chainSimulator"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	coreAPI "github.com/multiversx/mx-chain-core-go/data/api"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	defaultPathToInitialConfig = "../../../../node/config/"
+	sovereignConfigPath        = "../../../config/"
+	esdtSafeWasmPath           = "../testdata/esdt-safe.wasm"
+	feeMarketWasmPath          = "../testdata/fee-market.wasm"
+	multiSigWasmPath           = "../testdata/multisigverifier.wasm"
+)
+
+var oneEgld = big.NewInt(1000000000000000000)
+var initialMinting = big.NewInt(0).Mul(oneEgld, big.NewInt(100))
+var issueCost = big.NewInt(5000000000000000000)
+
+func TestBridge_DeployOnMainChain_IssueAndDeposit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck: false,
+		TempDir:                t.TempDir(),
+		PathToInitialConfig:    defaultPathToInitialConfig,
+		NumOfShards:            3,
+		GenesisTimestamp:       time.Now().Unix(),
+		RoundDurationInMillis:  uint64(6000),
+		RoundsPerEpoch: core.OptionalUint64{
+			HasValue: true,
+			Value:    20,
+		},
+		ApiInterface:                api.NewNoApiInterface(),
+		MinNodesPerShard:            1,
+		MetaChainMinNodes:           1,
+		MetaChainConsensusGroupSize: 1,
+		ConsensusGroupSize:          1,
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocksUntilEpochIsReached(3)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(0)
+	systemScAddress, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode("erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu")
+	systemEsdtScAddress, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode("erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u")
```

Comment:
Same remarks globally for
- shard id 0/const
- funcs to get addresses
- generating randomg blocks
- test naming conventions
- describe what each test is doing